src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringTranslator.cs

```diff
@@ -0,0 +1,431 @@
+        [NotNull] static readonly MethodInfo StartsWith = typeof(string).GetRuntimeMethod(nameof(string.StartsWith), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo IndexOfString = typeof(string).GetRuntimeMethod(nameof(string.IndexOf), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo IndexOfChar = typeof(string).GetRuntimeMethod(nameof(string.IndexOf), new[] { typeof(char) });
+        [NotNull] static readonly MethodInfo IsNullOrWhiteSpace = typeof(string).GetRuntimeMethod(nameof(string.IsNullOrWhiteSpace), new[] { typeof(string) });
+        [NotNull] static readonly MethodInfo Trim = typeof(string).GetRuntimeMethod(nameof(string.Trim), Type.EmptyTypes);
+        [NotNull] static readonly MethodInfo TrimWithChars = typeof(string).GetRuntimeMethod(nameof(string.Trim), new[] { typeof(char[]) });
+        [NotNull] static readonly MethodInfo TrimEndWithChars = typeof(string).GetRuntimeMethod(nameof(string.TrimEnd), new[] { typeof(char[]) });
+        [NotNull] static readonly MethodInfo TrimStartWithChars = typeof(string).GetRuntimeMethod(nameof(string.TrimStart), new[] { typeof(char[]) });
+        [NotNull] static readonly MethodInfo Substring = typeof(string).GetTypeInfo().GetDeclaredMethods(nameof(string.Substring)).Single(m => m.GetParameters().Length == 2);
+        [NotNull] static readonly MethodInfo Replace = typeof(string).GetRuntimeMethod(nameof(string.Replace), new[] { typeof(string), typeof(string) });
+
+        // The following exist as optimizations in netcoreapp20
+        [NotNull] static readonly MethodInfo TrimWithSingleChar = typeof(string).GetRuntimeMethod(nameof(string.Trim), new[] { typeof(char) });
+        [NotNull] static readonly MethodInfo TrimEndNoParam = typeof(string).GetRuntimeMethod(nameof(string.TrimEnd), new Type[0]);
+        [NotNull] static readonly MethodInfo TrimEndSingleChar = typeof(string).GetRuntimeMethod(nameof(string.TrimEnd), new[] { typeof(char) });
+        [NotNull] static readonly MethodInfo TrimStartNoParam = typeof(string).GetRuntimeMethod(nameof(string.TrimStart), new Type[0]);
+        [NotNull] static readonly MethodInfo TrimStartSingleChar = typeof(string).GetRuntimeMethod(nameof(string.TrimStart), new[] { typeof(char) });
+
+        #endregion
+
+        /// <summary>
+        /// The backend version to target.
+        /// </summary>
+        [CanBeNull] readonly Version _postgresVersion;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NpgsqlStringTranslator"/> class.
+        /// </summary>
+        /// <param name="postgresVersion">The backend version to target.</param>
+        public NpgsqlStringTranslator([CanBeNull] Version postgresVersion) => _postgresVersion = postgresVersion;
+
+        /// <inheritdoc />
+        [CanBeNull]
+        public virtual Expression Translate(MethodCallExpression e)
```

Comment:
Just a small suggestion - maybe check that the class is String and exit immediately otherwise? This is mainly to make it more clear that this translator is only about string.

Reply:
Good idea--updated.